lib/galaxy/datatypes/binary.py

```diff
@@ -4844,10 +4844,30 @@ def sniff(self, filename):
                 if len(header_bytes) != header_size:
                     return False
 
-                header = json.loads(header_bytes.decode("utf-8"))
+                # adapted from: https://github.com/galaxyproject/galaxy/blob/7a65393d4b2863ffd695a823a3f41da90bac9fee/lib/galaxy/datatypes/text.py#L108
```

Comment:
I would suggest to restructure the code, i.e. take the fucntion from text.py, move it to somewhere in https://github.com/galaxyproject/galaxy/tree/12bd9d350654f1d61a36f6e373bda8221943b696/lib/galaxy/datatypes/util  make both call this function.